DIPs/DIP1011.md

```diff
@@ -300,6 +300,34 @@ void function(Foo foo) y = x; // Error: cannot implicitly convert expression (x)
 
 This type of error already occurs with other linkage types so it should just work.
 
+## The & operator ambiguity
+
+Taking the address of a dotted member function (`&<object>.<member-function>`) creates a delegate.  The same syntax can be used for a UFCS call with no arguments (`&<first-argument>.<non-member-function>`); however, in this case it is interpreted as taking the address of the return value of a call to the function with no arguments e.g.
+
+```D
+struct Foo
+{
+    ref int bar1() { /* ... */ }
+}
+ref int bar2(ref Foo foo) { /* ... */ }
+
+Foo foo;
+auto a = &foo.bar1; // creates a delegate, type of a is ref int delegate()
+auto b = &foo.bar2; // calls function bar2 and takes the address of the return
+                    // value, type of b is int*
+```
+
+Note that even though the member function call could be interpreted in the same manner as the UFCS call, the '&' operator gives precedence to intepreting it as a delegate.  It is recommended that `extern(delegate)` functions behave the same way as member functions in this case for two reasons.  The first being that it maintains consistent syntax between member functions and `extern(delegate)` functions which makes it easier for templates/mixins to work the both kinds.  The second reason is that with this precedence, there currently exists syntax if the application really means to call the function and take the address of the return value, namely, by including an empty parameter list `()` e.g.
```

Comment:
"easier for templates/mixins to work the both kinds" => "easier for templates/mixins to work with both kinds"

"there currently exists syntax if the application  really means to call" => "there currently exists syntax that expresses the intent to call"